suite-native/staking/src/selectors.ts

```diff
@@ -0,0 +1,217 @@
+import { NetworkSymbol } from '@suite-common/wallet-config';
+import { Account, AccountKey } from '@suite-common/wallet-types';
+import { selectAccountByKey } from '@suite-common/wallet-core';
+
+import {
+    selectEthereumAccountHasStaking,
+    selectEthereumIsStakePendingByAccountKey,
+    selectVisibleDeviceEthereumAccountsWithStakingByNetworkSymbol,
+    selectEthereumIsStakeConfirmingByAccountKey,
+    selectEthereumAPYByAccountKey,
+    selectEthereumStakedBalanceByAccountKey,
+    selectEthereumRewardsBalanceByAccountKey,
+    selectEthereumTotalStakePendingByAccountKey,
+} from './ethereum/ethereumSelectors';
+import { NativeStakingRootState } from './types';
+import { doesCoinSupportStaking } from './utils';
+
+// create empty array in advance so it will be always same on shallow comparison
+const EMPTY_ACCOUNT_ARRAY: Account[] = [];
+
+export const selectDeviceAccountsWithStaking = (
+    state: NativeStakingRootState,
+    networkSymbol: NetworkSymbol,
+) => {
+    if (!doesCoinSupportStaking(networkSymbol)) {
+        return EMPTY_ACCOUNT_ARRAY;
+    }
+
+    switch (networkSymbol) {
+        case 'eth':
+        case 'thol':
+        case 'tsep':
+            return selectVisibleDeviceEthereumAccountsWithStakingByNetworkSymbol(state, 'eth');
+        default:
+            // This throws error if any networkSymbol is not handled.
+            networkSymbol satisfies never;
+
+            return EMPTY_ACCOUNT_ARRAY;
+    }
+};
+
+export const selectHasAnyDeviceAccountsWithStaking = (
+    state: NativeStakingRootState,
+    networkSymbol: NetworkSymbol,
+) => {
+    if (!doesCoinSupportStaking(networkSymbol)) {
+        return false;
+    }
+
+    switch (networkSymbol) {
+        case 'eth':
+        case 'thol':
+        case 'tsep':
+            return selectVisibleDeviceEthereumAccountsWithStakingByNetworkSymbol(state, 'eth');
```

Comment:
make sure to return bool for type safety.

```suggestion
            return A.isNotEmpty(selectVisibleDeviceEthereumAccountsWithStakingByNetworkSymbol(state, 'eth'));
```

Reply:
fixed